erpnext/deprecation_dumpster.py

```diff
@@ -109,3 +109,19 @@ def deprecation_warning(marked: str, graduation: str, msg: str):
 
 
 ### Party starts here
+@deprecated(
+	"erpnext.controllers.taxes_and_totals.get_itemised_taxable_amount",
+	"2024-11-07",
+	"v17",
+	"The field item_wise_tax_detail now already contains the net_amount per tax.",
+)
+def taxes_and_totals_get_itemised_taxable_amount(items):
+	import frappe
+
+	itemised_taxable_amount = frappe._dict()
+	for item in items:
+		item_code = item.item_code or item.item_name
+		itemised_taxable_amount.setdefault(item_code, 0)
+		itemised_taxable_amount[item_code] += item.net_amount
+
+	return itemised_taxable_amount
```

Comment:
This proposal to deprecate has two goals:
1. Prevent new usage of this method; nudging users to use the more reliable and stable means implemented by this PR
2. Eventually remove the liability and cost to maintain these lines of codes


The main goal is point 1, therefore, we also can also set the graduation version beyond v17.

A good graduation point would be something that matches the natural rate of change and evolution of code within the ecosystem.

Given that current downstream users have made use of brittle work around for the previous limitations, it might be a good thing to prompt them to revisit this code within the span of roughly 1-2 years.